image.py

```diff
@@ -0,0 +1,91 @@
+import cairo 
+from cairosvg import svg2png
+from io import BytesIO 
+
+# Yes going through cairo to a png and then back into cairo again
+# But all SVG libraries are hot garbage and most don't work at all
+# This one at least works, but doesn't render to an object, only a file
+# So we're stuck with the roundtrip.
+
+# Set of languages for which we have a flag
+# Should probably do this programmatically
+known_langs = {'hr', 'uk', 'fi', 'mk', 'hy', 'sv', 'bg', 'ru', 'hu', 'pl', 
+               'el', 'de', 'tr', 'sl', 'es', 'en', 'ca', 'pt', 'cs', 'nl', 
+               'ro', 'sq', 'sk', 'be', 'sr', 'fr', 'et', 'lv', 'it', 'da', 
+               'no', 'lt', 'az'}
+
+def generate_image(steps, filename):
+    # Set all the constants!
+    # Possibly a better way to achieve this?
+    top_margin = 75
+    left_margin = 100
+    right_margin = 150
+    bottom_margin = 100
+    step = 100
+
+    width = 800
+    height = step * len(steps) + bottom_margin
+
+    background_colour = (0, 0, 0)
+
+    font_size = 48
+    flag_height = 64 # We are dumb and don't read this from the file
+    flag_width = 64 # ditto
+
+    font_name = "Helvetica"
+    font_slant = cairo.FONT_SLANT_NORMAL
+    font_weight = cairo.FONT_WEIGHT_NORMAL
+    font_colour = (1, 1, 1)
+
+    text_indent = 36
+    text_hposition = left_margin + flag_width + text_indent
+    text_width = width - text_hposition - right_margin
+
+    image = cairo.ImageSurface(cairo.FORMAT_ARGB32, width, height)
+    context = cairo.Context(image)
+
+    # Create the background
+    context.set_source_rgb(*background_colour)
+
+    # Leave the last row transparent so Twitter doesn't JPEG us
+    context.rectangle(0, 0, width, height - 1)
+    context.fill()
+    context = cairo.Context(image)
+    
+    for i, (lang, title) in enumerate(steps):
+        if lang in known_langs:
+            flag = "flags/" + lang + ".svg"
+        else:
+            flag = "flags/unknown.svg"
+        flag = BytesIO(svg2png(url=flag, ))
+        flag = cairo.ImageSurface.create_from_png(flag)
+        context.set_source_surface(flag, left_margin, top_margin + step * i)
+        context.paint()
+        
+        # Here we recreate the context because Cairo is dumb
+        # (or because I don't understand it properly)
+        # Either way otherwise we get no text.
+        context = cairo.Context(image)
+        context.set_source_rgb(*font_colour)
+
+        # We will need to generalise this if the Yandex API ever supports CJK
```

Comment:
The API only supports a subset of the web languages though? Not just beta languages are omitted; the API only gives 33 languages while the web has 91, with 15 in beta.